afs/afs-ws/afs-ws-storage/src/main/java/com/powsybl/afs/ws/storage/NodeEventClient.java

```diff
@@ -52,8 +69,52 @@ public void onError(Throwable t) {
         }
     }
 
+
     @OnClose
-    public void onClose(Session session) {
-        LOGGER.trace("Node event websocket session '{}' closed for file system '{}'", session.getId(), fileSystemName);
+    public void onClose(Session session, CloseReason closeReason) {
+
+        if (closeReason.getCloseCode().getCode() == 1006) {
+            URI wsUri = null;
+            if (this.reconnectionInterval > config.getReconnectionMaxInterval()) {
+                LOGGER.warn("Node event websocket session '{}' closed for file system '{}'", session.getId(), fileSystemName);
+                return;
+            }
+            if (counter >= 1) {
+                this.reconnectionInterval = this.reconnectionInterval * config.getReconnectionIntervalMutiplier();
+            } else {
+                this.reconnectionInterval = config.getReconnectionInitialInterval();
+            }
+            WebSocketContainer container;
+            try {
+
+                sleep(reconnectionInterval * 1000L);
+                wsUri = RemoteListenableAppStorage.getWebSocketUri(config.getRestUri());
+                URI endPointUri = URI.create(wsUri + "/messages/" + AfsRestApi.RESOURCE_ROOT + "/" +
+                        AfsRestApi.VERSION + "/node_events/" + fileSystemName);
+                LOGGER.debug("Connecting to node event websocket at {}", endPointUri);
+                container = ContainerProvider.getWebSocketContainer();
+                container.connectToServer(this, endPointUri);
+                resetComptor();
+            } catch (DeploymentException e) {
+                if (e.getCause().getCause() instanceof ConnectException) {
+                    LOGGER.error(e.getCause().getCause().getMessage(), e);
+                    updateCounter();
+                    this.onClose(session, closeReason);
+                }
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                throw new UncheckedInterruptedException(e);
+            }
+        }
+    }
+
+    public static void updateCounter() {
+        counter = counter + 1;
+    }
+
+    public static void resetComptor() {
```

Comment:
Refactor to resetCounter

Reply:
The counter attribute is replaced by a local variable so the method resetComptor() is deleted